### PR TITLE
Rename functional tests dir and files

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
Prep for some more yea cleanups and new features.

Changes:

This will allow us to:
- remove explicit adding of "id" key in the yea spec file
- import files as we no longer start files with a number or have a hyphen
- The directory hierarchy will allow us to have composable tests,  for example there can be a top level 1_service that runs all the 0_main with service on
- Introduce dual ids, numbers `0.0.0` and names `main.core.blah` and a hybrid `0_main.0_core.0_blah'
- numbers are used so we can be sure to have consistent tracking of tests over time 

Not done:
- mp dir
- imports dir
- not all hyphen files were renamed with underscores they were only renamed if they already had test numbers.  If they didnt have test numbers they will be renamed in a subsequent pass when yea is updated

Coming soon:
- new yea that has some new features and can get rid of the unneeded id tags in each test

Testing
-------
I checked to make sure i got most files with:
```
find . -type f | egrep "/\d" | egrep -v "t\d" | grep -v /mp/ | grep -v /imports/
```

and that program entries were fixed:
```
find . -name \*.yea | xargs fgrep program | grep "/t" 
```

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

**Comment by dmitryduev** [comment]
Looks awesome! Could you please fix the assertions in these tests: https://app.circleci.com/pipelines/github/wandb/client/13803/workflows/2744e841-cefd-4090-8156-97072e68bb16/jobs/227522

File: tests/functional_tests/t0_main/debug/t5_sentry_third_party.yea [hunk 1]
```diff
@@ -6,7 +6,7 @@ tag:
     - linux
     - mac
 command:
-  program: sentry-third-party.py
+  program: sentry_third_party.py
```
Comment (dmitryduev) [inline tests/functional_tests/t0_main/debug/t5_sentry_third_party.yea:9]: `tests/functional_tests/t0_main/debug/sentry-third-party.py` needs to be renamed

File: tests/functional_tests/t0_main/lightning/train_tpu_ddp.yea [hunk 2]
```diff
@@ -5,7 +5,7 @@ tag:
   shard: service
   skip: true
 command:
-  program: train-tpu-ddp.py
+  program: train_tpu_ddp.py
```
Comment (dmitryduev) [inline tests/functional_tests/t0_main/lightning/train_tpu_ddp.yea:8]: Don't need this at all?

**Review by dmitryduev — APPROVED** [review]
very cool, thanks Jeff!
👍🏻 with a couple comments.

File: tests/functional_tests/t0_main/debug/t5_sentry_third_party.yea (hunk 1 above)
Comment (raubitsj) [reply tests/functional_tests/t0_main/debug/t5_sentry_third_party.yea:9]: fixed,. thanks

File: tests/functional_tests/t0_main/lightning/train_tpu_ddp.yea (hunk 2 above)
Comment (raubitsj) [reply tests/functional_tests/t0_main/lightning/train_tpu_ddp.yea]: fixed thanks

**Pull request merged**